src/NServiceBus.Core/Recoverability/Faults/StoreFaultsInErrorQueue.cs

```diff
@@ -18,7 +19,12 @@ protected internal override void Setup(FeatureConfigurationContext context)
         {
             var errorQueue = ErrorQueueSettings.GetConfiguredErrorQueue(context.Settings);
             context.Settings.Get<QueueBindings>().BindSending(errorQueue);
-            context.Pipeline.Register(new MoveFaultsToErrorQueueBehavior.Registration(errorQueue, context.Settings.LocalAddress()));
+
+            var failureInfoStorage = new FailureInfoStorage(context.Settings.GetOrDefault<int?>("Recoverability.FailureInfoStorage.CacheSize") ?? 1000);
+            context.Container.RegisterSingleton(failureInfoStorage);
```

Comment:
We need to share the store among all behaviors to be able to pass proper retry header values to custom retry policies.